src/raftstore/store/peer.rs

```diff
@@ -791,6 +786,17 @@ impl Peer {
         send_msg.set_from_peer(from_peer);
         send_msg.set_to_peer(to_peer);
 
+        if self.get_store().is_initialized() &&
+           (msg_type == MessageType::MsgRequestVote ||
+            // the peer has not been known to this leader, it may exist or not.
+            msg_type == MessageType::MsgHeartbeat && msg.get_commit() == INVALID_INDEX) {
```

Comment:
better to use `(msg_type == MessageType::MsgHeartbeat && msg.get_commit() == INVALID_INDEX)`